pkg/access_token/access_token_file_test.go

```diff
@@ -19,138 +19,200 @@ func TestAccessTokenFile(t *testing.T) {
 	var tokenInFile, _ = NewAccessTokenFile(config)
 
 	Convey("Read", t, func() {
-		Convey("Returns true if Data output as expected", func() {
+
+		Convey("Given an access token with data", func() {
 			dataActual := []byte{'t', 'e', 's', 't'}
 			err := tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
 
-			dataExpected, err := tokenInFile.Read()
-			eq := reflect.DeepEqual(dataActual, dataExpected)
+			Convey("Finishes without raising an error", func() {
+				So(err, ShouldEqual, nil)
+			})
+
+			Convey("When running Read method", func() {
+				dataExpected, err := tokenInFile.Read()
+
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
 
-			So(err, ShouldEqual, nil)
-			So(eq, ShouldEqual, true)
+				Convey("Returns the data that was written", func() {
+					eq := reflect.DeepEqual(dataActual, dataExpected)
+					So(eq, ShouldEqual, true)
+				})
+			})
 		})
 
-		Convey("Returns error if Data is nil", func() {
+		Convey("Given an access token's data is empty", func () {
 			tokenInFile.Data = nil
-			_, err := tokenInFile.Read()
 
-			So(err.Error(), ShouldEqual, "error reading access token, reason: data is empty")
+			Convey("When the token data is empty and running the Read method", func() {
```

Comment:
This can change to `When running Read method`. the prefix is already in the `given` clause